### DISCUSSION TITLE
Yet another "Symbol's function definition is void: project root" bug.

### DISCUSSION THREAD

**Discussion by derifatives** [joaotavora/eglot#980] [General]
I believe I've read all the relevant bugs on this, but not sure what to do. When I open up a file (in my case Haskell) and type `M-x eglot`, I always get "Symbol's function definition is void: project-root." I installed eglot via the standard `M-x package-install` route, and `M-x find-library` for `project` shows I've got project 0.8.1 under my elpa subdirectory.

Any ideas?

**Comment by joaotavora** [comment]
Perhaps you do have it in your elpa subdirectory, but is it loaded?  I think evidently not, otherwise you would have the function loaded.  You may have an earlier version of `project.el` shadowing it. Dunno, try `M-x list-load-path-shadows` and look for `project.el` in there.

**Comment by derifatives** [comment]
If I do `M-x list-load-path-shadows`, I have two relevant lines:
```
/home/rif/.emacs.d/elpa/project-0.8.1/project hides /usr/share/google-emacs/site-lisp/emacs-google-config/third_party/elisp/project/project
/home/rif/.emacs.d/elpa/project-0.8.1/project hides /usr/share/google-emacs/27.2+gg3+1.20210329.053400.rc147/lisp/progmodes/project
```

So it looks like the 0.8.1 is hiding the other versions. The `M-x find-library` finding 0.8.1 also seems to suggest it wants to load 0.8.1.  Any other thoughts?

**Reply by joaotavora** [reply]
It does seem you have some older version of project.el there.  Maybe you could just delete them?  Or at least make sure they aren't in the load path?  You can try `C-h f RET project-roots RET`, go to the `*Help*` buffer, follow the link and see where that lands you.

You haven't said which version of Emacs you're using and whether you're using a complex configuration or not, or if you've tried without said configuration.  Regardless, maybe just start with sth like `emacs -Q --package-initialize -L path/to/eglot -l eglot` recipe, and add little bits of your configuration until you find where you have a problem.
